src/main/java/eu/darkbot/popcorn/def/CaptchaPicker.java

```diff
@@ -104,13 +64,95 @@ public boolean canRefresh() {
 
     @Override
     public String status() {
-        return "Solving captcha" + (currentMatch != null ? " " + currentMatch : "");
+        return "Solving captcha" + (boxMatch != null ? " " + boxMatch : "");
     }
 
     @Override
     public void tick() {
+        boxes.stream()
+                .filter(box -> box.type.equals(boxMatch.name))
+                .min(Comparator.comparingDouble(box -> hero.locationInfo.now.distance(box)))
+                .ifPresent(box -> {
+                    if (!boxMatch.hasAmount && amountToCollect == 0) {
+                        amountToCollect = (int) boxes.stream()
+                                .filter(b -> b.type.equals(boxMatch.name))
+                                .count();
+                    } else if (boxMatch.hasAmount && amountToCollect == 0) {
+                        amountToCollect = boxMatch.amount;
+                    }
+                    collectBox(box);
+                });
+        if (currentlyCollected != 0 && currentlyCollected == amountToCollect) {
+            reset();
+            goBack();
+        }
```

Comment:
Shouldn't go back until it's actually cleared, and no need for reset.
Simply make sure that the list of boxes no longer contains any of the type of boxes (red nor black), then u can go back